index.js

```diff
@@ -122,21 +125,37 @@ Liftoff.prototype.buildEnvironment = function (opts) {
   }
 
   // use rechoir to autoload any required modules
-  var autoloads;
-  if (configPath) {
-    autoloads = rechoir.prepare(this.extensions, configPath, cwd, true);
-    if (autoloads instanceof Error) {
-      autoloads = autoloads.failures;
-    }
-    if (Array.isArray(autoloads)) {
-      autoloads.forEach(function (attempt) {
-        if (attempt.error) {
-          this.emit('requireFail', attempt.moduleName, attempt.error);
-        } else {
-          this.emit('require', attempt.moduleName, attempt.module);
-        }
-      }, this);
+  var registerLoader = function(extensions, configPath) {
+    var autoloads;
+    if (configPath) {
+      autoloads = rechoir.prepare(extensions, configPath, cwd, true);
+      if (autoloads instanceof Error) {
+        autoloads = autoloads.failures;
+      }
+      if (Array.isArray(autoloads)) {
+        autoloads.forEach(function (attempt) {
+          if (attempt.error) {
+            this.emit('requireFail', attempt.moduleName, attempt.error);
+          } else {
+            this.emit('require', attempt.moduleName, attempt.module);
+          }
+        }, this);
+      }
     }
+  }.bind(this);
+
+  registerLoader(this.extensions, configPath);
+
+  var configFiles = {};
+  if (isPlainObject(this.configFiles)) {
+    var fined = new Fined(cwd, this.extensions);
```

Comment:
Cool, since it isn't an issue, implement it with a pure function instead of the constructor pattern and if we need to optimize it later, we can dig into that separately.